### DISCUSSION TITLE
Suggestion: Improving slider velocity readability

### DISCUSSION THREAD

**Discussion by ghost** [ppy/osu#13729] [Ideas]
Simply put, I would love for snaking sliders to snake in at the speed that you would play the note, as opposed to the speed of the snaking being solely decided by a slider's length. That's what I thought the whole point of choosing snaking over regular was, but some editor testing quickly showed against my initial theory.

If this were to be implemented, it could allow for mappers to have more freedom in complex slider velocity and BPM's, as the slider will show the velocity beforehand. Major changes in velocity won't simply be a "surprise, you dropped a slider" moment anymore.

On a more extreme mapping level of the prior point would be accelerating/decelerating sliders actually being playable, but that's dependent on how easily readable an accelerating snaking slider is.

I particularly would love it as there are sometimes notes or complete shifts in song speed that cause me to drop combo simply through being out of the blue on a sight-read, where this could fix that

**Comment by LittleEndu** [comment]
> snake out at the speed that you would play the note

You meant "snake in". The slider snakes into position before you need to click it and snakes out of position as you play the note.

Otherwise it's a good idea, but I doubt this would enable mappers to map more complex SVs unless you can show an example of such a map. The SV would still need to be supported by the song and if it's supported by the song it can be anticipated by the player, snaking speed or not.

**Comment by mk56-spn** [comment]
i think regardless of whether this gets done snake in needs its slider speed lowered. right now its WAY too fast to the point where its often not noticeable

**Comment by Doksha** [comment]
Maybe as a mod, since this could allow for a small subset of maps to be much easier (maze maps and SV maps).
I agree that this would be a very cool feature, I use it when rendering with danser (replay renderer) and it looks so much better than the current implementation.
A setting would definitely be much nicer though.

**Comment by Fivoka** [comment]
As this idea aims to make slider velocities more readable, I suggest renaming the discussion post to reflect that better, or having my reply be moved to a new discussion post.

### Issues from previous replies

There is a huge problem with using a slider's snake-in animation to represent its SV, which is that a slow slider will barely show any of its slider body. And in some extremely slow cases, making it potentially misreadable as one circle. On top of that, every single slider art would get ruined because of this.

Making it optional through a mod also goes against trying to fix this issue.

### Suggested solution

What I suggest then, is to show a slider's velocity with a faint white highlight going through the slider's body or outline (or both).

Having the SV highlight move through a slider's outline presents some problems though. Mainly that a slider outline's color can be any color possible, while a slider body's color is always dark or slightly darker than the combo-color being used, even when that combo-color is white, making it much easier to avoid having the highlight blend in with the slider.

Hopefully, this serves as a good starting point for discussing ideas on how to make slider velocities more readable.
_I'd love to attach images to this reply or even work on a proof-of-concept ruleset for this but I think I'll let someone else to beat me to it._

**Reply by ghost** [reply]
Thanks for replying to my ancient post Fivoka.

I agree, having been 2.5 years since my initial post I can totally see those flaws in my old opinion. 

Having the main slider body stay with a slider velocity highlight going through it probably would be the best way to solve all of my initial problems without creating one's that the current system is looking to avoid. 

Your worry about highlights being invisible based on the combo color could be remedies by having the highlight color and potency be a skinnable element or setting it based on the slider body color (I prefer the former conceptually).

I don't know two things about how the game is coded so i can't really make a ruleset for it, but i can work an editor and animating software so i can probably make a video of what it can reasonably look like, though it'd take some time due to my ineptitude with both.

**Reply by 120-cell** [reply]
The issue of slider art being ruined could be solved by not snaking out sliders, as in this video https://www.youtube.com/watch?v=sDmhTU9ytrk, and the issue of misreading slow sliders as circles by visually distinguishing hitcircleoverlay.png and sliderstartcircleoverlay.png in the skin.
